packages/core/test/insured-bridge/BridgeRouter.js

```diff
@@ -53,27 +74,69 @@ describe("BridgeRouter", () => {
         return ev.l2DepositContract === newDepositContract;
       });
     });
-    it("Add whitelisted Token mapping", async () => {
-      const l1Token = rando;
-      const l2Token = owner;
-      const bridgePoolAddress = l1CrossDomainMessenger;
-      assert(
-        await didContractThrow(
-          bridgeRouter.methods.whitelistToken(l1Token, l2Token, bridgePoolAddress).send({ from: rando })
-        ),
-        "OnlyOwner modifier not enforced"
-      );
+    describe("Whitelist tokens", () => {
+      it("Basic checks", async () => {
+        assert(
+          await didContractThrow(
+            bridgeRouter.methods
+              .whitelistToken(l1Token, l2Token, bridgePoolAddress, defaultGasLimit)
+              .send({ from: rando })
+          ),
+          "OnlyOwner modifier not enforced"
+        );
+
+        // Fails if depositContract not set in BridgeRouter
+        assert(
+          await didContractThrow(
+            bridgeRouter.methods
+              .whitelistToken(l1Token, l2Token, bridgePoolAddress, defaultGasLimit)
+              .send({ from: owner })
+          ),
+          "Deposit contract not set"
+        );
+        await bridgeRouter.methods.setDepositContract(depositBoxImpersonator).send({ from: owner });
 
-      const txn = await bridgeRouter.methods.whitelistToken(l1Token, l2Token, bridgePoolAddress).send({ from: owner });
-      await assertEventEmitted(txn, bridgeRouter, "WhitelistToken", (ev) => {
-        return ev.l1Token === l1Token && ev.l2Token === l2Token && ev.bridgePool === bridgePoolAddress;
+        // Successful call
+        await bridgeRouter.methods
+          .whitelistToken(l1Token, l2Token, bridgePoolAddress, defaultGasLimit)
+          .send({ from: owner });
       });
+      it("Add token mapping on L1 and sends xchain message", async () => {
+        await bridgeRouter.methods.setDepositContract(depositBoxImpersonator).send({ from: owner });
+        const whitelistTxn = await bridgeRouter.methods
+          .whitelistToken(l1Token, l2Token, bridgePoolAddress, defaultGasLimit)
+          .send({ from: owner });
+        const whitelistCallToMessengerCall = l1CrossDomainMessengerMock.smocked.sendMessage.calls[0];
```

Comment:
might be useful to check `l1CrossDomainMessengerMock.smocked.sendMessage.calls.length==1`

Reply:
Great point